Новые карты ЦК, рефактор шаттлов, тандердом

Furukia opened the pull request:
## What Does This PR Do
* Центральное командование НТ за авторством He11Force
* Передовая база Синдиката(ЦК синди) за моим авторством
* Изменённая зона Сол-торговцев и админ рума за авторством He11Force
* Починила тандердом. Дизайн тандердома за авторством: He11Force
* Убрала 3 ненужные кнопки выдачи ЕРТ разных оружеек в Secrets 
* Починила сломанные текстурки голодеков
* Разделила z2 на два уровня CentComm и Admin_zone. Первый - ИЦ зона, там спавны антагов, ЦК НТ и Синдиката, Экран лобби и Транзит зоны шаттлов. Второй почти полностью пустой Z-Lvl с админ румой и тандердомом, для админских эксперементов.

* А так же рефактор шаттлов
	* Убрала simulated/shuttle тип, вместе практически везде поменялись стены шаттлов на новый тип, полы шаттлов. А окна вообще были удалены и заменены уже существующими нормальными окнами для шаттлов.
	* Переделала все стены шаттлов через автоматическую smooth систему
	* Переделала добрую половину шаттлов. Где то по мелочи, где то полностью.
	* Переименовала загадочный `floor4` в `objective_check` чтобы сразу было понятно, что этот пол важен. (Используется для некоторых целей, это пол СБ на шаттле эвакуации)
	* Удалила нафиг кучу ненужных transit турфов, оставив лишь два для полётов: Юг-Север и Запад-Восток 
	* Добавила декалям флаг `do_not_delete_me` для того, чтобы они не пропадали с шаттлов если включить флаг (Метод костыльный, но лучше не придумала.)
	* Перенесла Лут торговцев в Код вместо инстансов...
	* Создала спавнеры для окон шаттлов (Сейчас существует проблема из-за которой окна шаттлов заспавненные ими пропускают воздух... Причина не ясна, но поэтому я их в итоге не применяла... Другие окна не сломались.)
	* Переделала расстановку иконок в файле shuttle.dmi эффективно разделив его на несколько тематических файлов и отдельные файлы для гладких стен шаттлов (Серые, Белые, Ниндзя)
	
Вроде бы я упомянула все изменения. Если найдёте проблемы в коде, при тестах и т.д. Сообщите. Дизайнерские идеи, не работающие на ЦК кнопки и т.д. - тоже можете писать мне.
## Изображения изменений
![ЦК_синди](https://user-images.githubusercontent.com/67807868/204380775-aa00509a-ab99-49a6-9009-980b88fd1433.png)
![ЦК_НТ](https://user-images.githubusercontent.com/67807868/204380828-69fcc912-89a2-4cc8-82a7-7ce262e73b49.png)
![Admin_shuttle1](https://user-images.githubusercontent.com/67807868/204380967-abe07bf9-fd0e-4eb4-8bf6-cb7d7d29c84c.png)
![Admin_shuttle2](https://user-images.githubusercontent.com/67807868/204380973-34c265e1-8553-429e-98aa-74144bd47e10.png)
![Admin_shuttle3](https://user-images.githubusercontent.com/67807868/204380975-83f7c3ec-8659-4588-a7b4-a616567146a6.png)
![Evacuation_shuttle1](https://user-images.githubusercontent.com/67807868/204380979-3f1394ff-e61b-4985-9cf5-fb15eb99126d.png)
![Evacuation_shuttle2](https://user-images.githubusercontent.com/67807868/204380980-4929c767-7583-48cf-b917-2024748968e6.png)
![Evacuation_shuttle3](https://user-images.githubusercontent.com/67807868/204380985-6b682440-a0f5-49d2-8326-83569d88d4fd.png)
![Evacuation_shuttle4](https://user-images.githubusercontent.com/67807868/204380988-7450bec4-7d75-41f5-b65a-85844b8167d3.png)
![Evacuation_shuttle5](https://user-images.githubusercontent.com/67807868/204380993-3131fa08-1c68-4049-bc06-1eceae563d8a.png)
![Evacuation_shuttle6](https://user-images.githubusercontent.com/67807868/204380995-a884972f-5f14-411d-a69e-379e95baee99.png)
![Evacuation_shuttle7](https://user-images.githubusercontent.com/67807868/204380996-703d63e1-8257-417a-83bb-1ad5830dfad6.png)
![Nuclear_shuttle](https://user-images.githubusercontent.com/67807868/204380999-f8fdc671-a134-47f3-bd6e-d4f0cbb669dd.png)
![SOL_shuttle](https://user-images.githubusercontent.com/67807868/204381001-7b67bc49-c43a-4b0f-b071-ac6d69d8dd81.png)

# Примеры визуала стен шаттла в игре
![2022-11-25_11-58-37](https://user-images.githubusercontent.com/67807868/204381329-ec3cf117-5fed-4fec-8a7a-ad13bfe7c137.png)
![2022-11-25_12-05-34](https://user-images.githubusercontent.com/67807868/204381575-fa5f1bfa-1571-4f6b-963f-c8d46e6993ae.png)
![2022-11-25_12-04-44](https://user-images.githubusercontent.com/67807868/204381579-6f9d5632-efe1-41f3-b7d7-d826090b7ccc.png)

Furukia committed:
Сквош всех коммитов

Furukia committed:
Перенос старого z2 в generic папку и переименование

Furukia committed:
Убираю лишний мусор плюс мелкие правки

Furukia committed:
Фикс Конфликтов

dj-34 reviewed (commented):
**Проверил, частично протестил. Явных проблем не обнаружил, свои пожеланию высказал автору еще в дискорде, но продублирую здесь.**

1. Линтеры ругаются - нужно пофиксить, заменив путь в линтере до z2/цк/админзоны.

2. Пофиксить ошибки при компиляции.

3. Маршрутка с полами - Pure Evil, imho.

![image](https://user-images.githubusercontent.com/20109643/204464855-0c0e28b4-0e8e-45e0-9ed5-fa408d7bdab6.png)

4. Fake space стоило бы заменить на стеклянный пол, или что-нибудь другое.

![image](https://user-images.githubusercontent.com/20109643/204464691-561c840c-5078-41e0-9f75-4bd0633d4ff9.png)

5. ЦК мне показалось в целом неплохим, но менее продуманым и, возможно, удобным чем предыдущее версия, выражается это в:
 - По прибытию приходится далековато бежать до чилл зоны и не совсем очевидно куда, сквозь двери, толпу и лужи прям возле дверей.
 - Центральная рубка/зона управления шатлом/палуба капитана находится не в центре - это грустно.
 - БСА расположено в жопе, сбоку шаттла ЦК. Выглядит нелогично.
 - Раньше здесь были оверлеи, которые были некорректны по коду, понимаю что затычка, в будущем стоит вернуть.
 
![image](https://user-images.githubusercontent.com/20109643/204465761-9bd80a7e-4c73-4d40-ab05-11d55f5692a0.png)

6. У Диско шара есть immovable вариант с anchored 1 по дефолту

![image](https://user-images.githubusercontent.com/20109643/204465882-d24e0830-bbd7-4b97-9308-2db4d6d4386b.png)

7. Автор сообщил, что есть проблематика с спавнерами окон шаттлов. Они пропускают сквозь себя воздух. В будущем стоит это решить.

8. У некоторых шаттлов виден левый край карты при перелете. Стоит их подвинуть правее.

![image](https://user-images.githubusercontent.com/20109643/204466109-3a1a5e87-e6f2-4fa1-b006-5d83b16bca22.png)

_Если найду что-то еще, то дополню новым коментарием. Все что написано выше, является скорее личным предпочтением, чем явная ошибка, за исключением проблем линтеров._

dj-34 commented:
8. Все аирлоки на ЦК не закрыты, так что можно без проблем выйти в космос в любом месте.

![image](https://user-images.githubusercontent.com/20109643/204471775-24886d2a-33d1-4f5f-9ebb-7c6edd02efc4.png)

9. Бардак с дир. стеклами, стоило бы их выравнить и сделать более симметричными.

![image](https://user-images.githubusercontent.com/20109643/204472222-8ffc7d6b-8090-4f40-b0ea-97bd58dc374e.png)

dj-34 commented:
10. Вижу была попытка (или уже починен) доработать/починить забор. Однако заметил что он пропал с объектов, а также есть ошибки в названии спрайта (fecnce).
![image](https://user-images.githubusercontent.com/20109643/204479717-76e707e3-63e8-4160-ba82-88ef0eb7c2f5.png)

Furukia committed:
Апдейт линтера.

Furukia committed:
Конвертирую карты в TGM формат

Furukia committed:
Возвращение старых иконок заборов

Во время разработки синди-ЦК я хотела создать зону для прогулок у тюрьмы синдиката окружив её забором. Попросила Эла нарисовать мне более склеиваемые заборы. Но потом у нас в игре уже добавили другие неплохие заборчики и я просто использовала их не став парится с кодом. А иконки Эла удалить забыла... Вот удаляю теперь. Может если у меня будет время, сделаю потом эти заборы отдельным ПР-ом. Но сомневаюсь в этом...

Furukia committed:
Фикс конфликтов с шаттлом големов

Furukia committed:
Фикс проблем карт и крестовый поход на двойные турфы

* Удалила все двойные турфы на 1 тайле
* Удалила все двойные латисы на 1 тайле
* Удалила все двойные столы, стулья на 1 тайле
* Поправила различные проблемы с кнопками на картах ЦК
* Починила сломавшуюся точку для прибытия шаттла нюкеров на дельте
* Добавила статую Мунивёрса на синди-ЦК по его просьбе
* Сделала неуязвимый вид шаттеров и применила его в разных местах на ЦК
* Обезопасила большинство кнопок на обоих ЦК от боргов.
* Сдвинула все шаттлы и их транзит зоны примерно на 2-3 клетки, чтобы не были слишком близко к границе.
* Некоторые правки доступа на ЦК НТ чтобы ЕРТ не обворовывали СРТ и т.д.
* Возможно ещё какие мелочи которые я не могу вспомнить...

Furukia committed:
Крестовый поход на зоны

* Заменила все зоны в которых отредактированы var-ы на нормально прописанные со всем необходимым в КОДЕ зоны.

Furukia committed:
Статуя муни и мелкие правки на тайпане

* По просьбе муни пока что переименовала и заменила описание на статуе
* Починила проблемы с доступами у инженеров тайпана, в паре мест, на которые мне сегодня показали...
* Починила название зон операционных на тайпане

dj-34 commented:
![image](https://user-images.githubusercontent.com/20109643/205385962-a085b78f-b4f8-4302-b25e-3a25f29371fc.png)

Furukia commented:
> ![image](https://user-images.githubusercontent.com/20109643/205385962-a085b78f-b4f8-4302-b25e-3a25f29371fc.png)

Возможно это потому, что у нас изначально выбрана кибериада в билде, а не дельта и её лаваленд?

dj-34 commented:
> > ![image](https://user-images.githubusercontent.com/20109643/205385962-a085b78f-b4f8-4302-b25e-3a25f29371fc.png)
> 
> Возможно это потому, что у нас изначально выбрана кибериада в билде, а не дельта и её лаваленд?

Нет, просто у тебя не правильно прописаны пути до карт с учётом регистра.

dj-34 commented on _maps/delta.dm:
```diff
@@ -12,16 +12,18 @@ Remapped by ThaumicNik, TrashDoxx, J4.BA, BeepBoop, mr_g, IceGreen, AlexRavenidz
+	#include "map_files\delta\delta.dmm"
+	#include "map_files\generic\CentComm.dmm"
+	#include "map_files\generic\Admin_Zone.dmm"
+	#include "map_files\delta\lavaland.dmm"
```
```suggestion
	#include "map_files\Delta\Lavaland.dmm"
```

dj-34 commented on _maps/delta.dm:
```diff
@@ -12,16 +12,18 @@ Remapped by ThaumicNik, TrashDoxx, J4.BA, BeepBoop, mr_g, IceGreen, AlexRavenidz
-	#include "map_files\Delta\delta.dmm"
-	#include "map_files\cyberiad\z2.dmm"
-	#include "map_files\Delta\Lavaland.dmm"
+	#include "map_files\delta\delta.dmm"
```
```suggestion
	#include "map_files\Delta\delta.dmm"
```

Furukia committed:
Apply suggestions from code review

Co-authored-by: dj-34 <dj-34@bk.ru>

Furukia committed:
Более новая версию спрайта для статуи муни

dj-34 reviewed (approved):
Частично проверил в игре и в редакторе. Явных багов не заметил, из недочётов нашёл только реверснутый PARALLAX_MOVEDIR у шаттла торговцев и вессела НТ - в следствии чего они летают жопой.
Остается только мержить, собирать и отрабатывать отзывы.

PhantornRU commented:
Все отсылки на английском, а старые на русском :(

Когда-нибудь мы сделаем эти бронированные стены более красивыми... Когда-нибудь... Но не сегодня :(
